src/shared/remoteSourceTrackingService.ts

```diff
@@ -604,15 +526,58 @@ function getDecodedKeyIfSourceMembersHas({
   sourceMembers,
   logger,
 }: {
-  sourceMembers: Dictionary<MemberRevision>;
+  sourceMembers: Map<string, MemberRevision>;
   key: string;
   logger: Logger;
 }): string {
   const originalKeyDecoded = decodeURIComponent(key);
-  const match = Object.keys(sourceMembers).find((memberKey) => decodeURIComponent(memberKey) === originalKeyDecoded);
+  const match = Array.from(sourceMembers.keys()).find(
+    (memberKey) => decodeURIComponent(memberKey) === originalKeyDecoded
+  );
   if (match) {
     logger.debug(`${match} matches already tracked member: ${key}`);
     return match;
   }
   return key;
 }
+
+const getFilePath = (orgId: string): string => path.join('.sf', 'orgs', orgId, FILENAME);
+
+const readFileContents = async (filePath: string): Promise<Contents | Record<string, never>> => {
+  try {
+    const contents = await fs.promises.readFile(filePath, 'utf8');
+    return parseJsonMap<Contents>(contents, filePath);
+  } catch (e) {
+    return {};
```

Comment:
I think we should at least log the error, if not handle the errors and throw more meaningful ones.  Returning `{}` doesn't satisfy the return type does it?
When this function is called, the file is expected to be there and have parseable content so this shouldn't ever happen but if it does we should know the reason why, versus continue silently with empty state.

Reply:
logged.  It's not necessarily an error if there is no file there yet to read.  

The type can be an empty {} (that's the Record<string, never> alternative)

Reply:
alternatively, we could change it to catch the ENOENT and re-throw other things?